### ISSUE TITLE
Problems when installing using pip

### ISSUE THREAD

**Issue by set-soft** [Technologicat/mcpyrate#17]
A user reported the following in my project: https://github.com/INTI-CMNB/KiBot/issues/29

The problem was related to the way mcpyrate looks for the module paths: https://github.com/INTI-CMNB/KiBot/pull/30

When installing `kibot` using `pip install --user` the `sys.path` variable contains:

```
/home/user/.local/
```

and

```
/home/user/.local/lib/python3.8/site-packages/
```

A submodule from `/home/user/.local/lib/python3.8/site-packages/` will match (startswith) `/home/user/.local/` before matching the real path. 

The patch provided by @skorokithakis fixes this problem. 

I think it should be part of `mcpyrate`.

**Comment by Technologicat** [comment]
Yes, my mistake, seems I didn't consider this case. Python imports can be tricky to get right. Curiously, I have `mcpyrate` itself installed as `--user` (in CPython 3.6), but I hadn't seen this error.

I think the obviously correct way to sort the paths is to count the number of path separators, then reverse-sort on that (which fixes the problem - we want subdirectories before their parents), and as a secondary key, forward-sort the names lexicographically (for readability, if someone looks at the list).

It may be that reverse-sorting on string length results in the correct behavior, but strictly speaking that requires a proof. ;)

Thanks for reporting! I'll fix this in `mcpyrate` ASAP.

**Comment by Technologicat** [comment]
(The required proof is that for any given directory path, represented as a string, the string representation of any of its parent directories is always a prefix of the original string. Hence the parent directories are represented by strings shorter than the original. Symbolic links don't matter here, since we're operating at a level of abstraction where we should consider them on equal footing with real directories.

Still, I think it's clearer (hence requiring less explanation) if we make it explicit that the important property is the number of path separators, not the string length. Consider a case with `a/b/c` and `foo/bar` - the *longer string first* rule looks first in `foo/bar`. These directories aren't related, so it doesn't matter, but it may leave the reader puzzled as to why that particular lookup ordering.)

**Comment by Technologicat** [comment]
This should do it (in `mcpyrate/coreutils.py`, function `match_syspath`):

```python
    def sortkey(s):  # Match deeper paths first; for readability, break ties lexicographically.
        return -s.count(os.path.sep), s
    for root_path in sorted(sys.path, key=sortkey):
        ...
```

I'll test and push it probably later today.

**Comment by skorokithakis** [comment]
Personally, I agree with the proof but also have no issue with the improved fix, thank you!

**Comment by Technologicat** [comment]
Fix pushed to `mcpyrate`, see 6f1bbef.
